### ISSUE TITLE
Import/Export Functionality

### ISSUE THREAD

**Issue by d4mation** [realbigplugins/client-dash#159]
This could be useful for site owners trying to keep Local/Staging/Live in sync, but the primary benefit I can see for this would be support.

Users could post their Export File to the WP.org Support Ticket and for Client Dash Pro, it could be automatically included as part of RBP Support.

**Comment by joelworsham** [comment]
This one is stupid simple too. I'd put an interface on the Settings page for import/export. They can select a role (or all roles) from a dropdown, then click a button and they get a json file of the customizations. Maybe even a checkbox for selecting menus and/or dashboard widgets.

**Comment by d4mation** [comment]
I figure it could even be generated as a Base64 Encoded Data URI for the download and attached to the RBP Support Email in the same way as the regular Debug File thereby never needing to actually _store_ the file anyway and also managing to re-use the method as well :)

**Comment by joelworsham** [comment]
We could do that, but I absolutely want to have the user download a JSON, because that would be the primary use of the export/import feature.

**Comment by d4mation** [comment]
Yeah, that should be possible using a Base64 Encoded Data URI. You can serve any file type that way actually.

There's a bunch of ways to do it and HTML5 makes it easier. There's a few libraries out there too.

https://stackoverflow.com/questions/3665115/create-a-file-in-memory-for-user-to-download-not-through-server

**Comment by d4mation** [comment]
In this case, it'd probably be pure JS rather than an HTML Attribute though. That way it can use returned data from the server dynamically based on their choices on what to export.
